core/src/main/java/org/mapfish/print/http/HttpCredential.java

```diff
@@ -98,7 +98,11 @@ public void validate(final List<Throwable> validationErrors, final Configuration
     public boolean matches(final MatchInfo matchInfo) throws SocketException, UnknownHostException, MalformedURLException {
         for (URIMatcher uriMatcher : this.getMatchers()) {
             if (uriMatcher.matches(matchInfo)) {
-                return true;
+                if (uriMatcher.isReject()) {
```

Comment:
The Checkstyle linter (and me as well) prefers:

    return !uriMatcher.isReject();